src/components/implementation/resmgr/naive/res_mgr.c

```diff
@@ -0,0 +1,324 @@
+}
+
+thdcap_t
+resmgr_initthd_create_intern(spdid_t cur, spdid_t s)
+{
+	struct cos_defcompinfo *res_dci = cos_defcompinfo_curr_get();
+	struct cos_compinfo *res_ci     = cos_compinfo_get(res_dci);
+	struct res_comp_info *rc = res_info_comp_find(cur);
+	struct res_comp_info *rs = res_info_comp_find(s);
+	struct sl_thd *t = NULL, *rt = NULL, *rst = NULL;
+	int ret;
+
+	assert(rc && res_info_init_check(rc));
+	assert(rs && res_info_init_check(rs));
+	assert(res_info_is_sched(cur) && res_info_is_child(rc, s));
+	assert(!res_info_is_sched(s));
+
+	t = sl_thd_child_initaep_alloc(res_info_dci(rs), 0, 0);
+	assert(t);
+	rt = res_info_thd_init(rc, t);
+	assert(rt);
+	rst = res_info_initthd_init(rs, t);
+	assert(rst);
+
+	/* child is not a scheduler, don't copy into child */
+	/* parent only needs the thdcap */
+	ret = cos_cap_cpy(res_info_ci(rc), res_ci, CAP_THD, sl_thd_thdcap(rt));
+	assert(ret > 0);
+	rs->p_initthdcap = ret;
+	rs->initthdid    = rt->thdid;
+
+	return ret;
+}
+
+thdcap_t
+resmgr_initaep_create_intern(spdid_t cur, spdid_t s, int owntc, int u1, asndcap_t *sndret, u32_t *rcvtcret)
+{
+	struct cos_defcompinfo *res_dci = cos_defcompinfo_curr_get();
+	struct cos_compinfo *res_ci     = cos_compinfo_get(res_dci);
+	struct res_comp_info *rc = res_info_comp_find(cur);
+	struct res_comp_info *rs = res_info_comp_find(s);
+	struct sl_thd *rt = NULL, *rst = NULL, *t = NULL, *rinit = NULL;
+	tcap_t tc;
+	arcvcap_t rcv;
+	int ret;
+
+	assert(rc && res_info_init_check(rc));
+	assert(rs && res_info_init_check(rs));
+	assert(res_info_is_sched(cur) && res_info_is_child(rc, s));
+	assert(res_info_is_sched(s));
+
+	rinit = res_info_initthd(rc);
+	assert(rinit);
+	t = sl_thd_ext_child_initaep_alloc(res_info_dci(rs), rinit, 1);
+	assert(t);
+	rt = res_info_thd_init(rc, t);
+	assert(rt);
+	rst = res_info_initthd_init(rs, t);
+	assert(rst);
+
+	/* child is a scheduler.. copy initcaps */
+	ret = cos_cap_cpy_at(res_info_ci(rs), BOOT_CAPTBL_SELF_INITTHD_BASE, res_ci, sl_thd_thdcap(rt));
+	assert(ret == 0);
+	ret = cos_cap_cpy_at(res_info_ci(rs), BOOT_CAPTBL_SELF_INITRCV_BASE, res_ci, sl_thd_rcvcap(rt));
+	assert(ret == 0);
+	if (owntc) {
+		ret = cos_cap_cpy_at(res_info_ci(rs), BOOT_CAPTBL_SELF_INITTCAP_BASE, res_ci, sl_thd_tcap(rt));
+		assert(ret == 0);
+	} else {
+		/* if it's a scheduler.. use parent's tcap (current spdid) */
+		ret = cos_cap_cpy_at(res_info_ci(rs), BOOT_CAPTBL_SELF_INITTCAP_BASE, res_ci, sl_thd_tcap(rinit));
+		assert(ret == 0);
+	}
+
+	/* parent needs tcap/rcv to manage time. thd/asnd to activate. */
+	ret = cos_cap_cpy(res_info_ci(rc), res_ci, CAP_THD, sl_thd_thdcap(rt));
+	assert(ret > 0);
+	rs->p_initthdcap = ret;
+	rs->initthdid    = rt->thdid;
+	rcv = cos_cap_cpy(res_info_ci(rc), res_ci, CAP_ARCV, sl_thd_rcvcap(rt));
+	assert(rcv > 0);
+	tc = cos_cap_cpy(res_info_ci(rc), res_ci, CAP_TCAP, sl_thd_tcap(rt));
+	assert(tc > 0);
+	*rcvtcret = (rcv << 16) | (tc);
+	*sndret = cos_cap_cpy(res_info_ci(rc), res_ci, CAP_ASND, sl_thd_asndcap(rt));
```

Comment:
I'm missing something here.  Looks like we're creating the initial rcv/tcap/thd for a boot capability image.  So now why are we also sending some form of `asnd`?

I would think that creating a rcv capability (with thd and tcap) is one operation.  Creating a send is another different operation.